### PR TITLE
Sy 1021 improved writer documentation

### PR DESCRIPTION
**Pull request by emilbon99**
# Fix Pull Request Template

## Key Information

- **Linear Issue**: [SY-1021](https://linear.app/synnaxlabs/issue/SY-1021/improved-writer-documentation)

## Description

Improves documentation pages related to writing data, including the writer concepts and the write data pages in both the Python and Typescript clients.

## Basic Readiness Checklist

- [x] I have performed a self-review of my code.
- [x] I have added sufficient regression tests to cover the changes to CI.
- [x] I have added relevant tests to cover the changes or exposing bugs.
- [x] I have verified code coverage targets are met.

## Additional Notes
- [ ] These changes deal with concurrency.
- [ ] These changes affect UI.

## Manual QA Additions

- [x] I have updated the [Release Candidate](/.github/PULL_REQUEST_TEMPLATE/rc.md) template
  with necessary manual QA steps to test my change.

## Reviewer Checklist
- [ ] Sufficient test coverage of new additions.
- [ ] Verified all steps in readiness checklists.
- [ ] UI changes have been tested.
- [ ] style and formatting is consistent.
- [ ] Reviewed any relevant changes to concurrent code for safety. 
- [ ] Sufficient comments and clarity of code.

### CODE REVIEW

**Comment by Lham42** [comment]
If you made changes within the python client - also indicate those in the pr description

File: docs/site/src/pages/reference/console/ui-overview.mdx [hunk 1]
```diff
@@ -53,7 +53,7 @@ workspaces.
     </tr>
     <tr>
       <td>[Tasks](/reference/device-drivers)</td>
-      <td>Tasks are the current read and write tasks happening on devices connected to
+      <td>Tasks are the current read and write tasks running on devices connected to
```
Comment (Lham42) [inline docs/site/src/pages/reference/console/ui-overview.mdx:56]: for all the descriptions in this table - do we need the "x are" at the front of each desciption?

File: docs/site/src/pages/reference/concepts/writes.mdx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+   </tbody>
+</Table>
+
+### Write Phase
+
+After all opening validation checks pass, the transaction enters the write phase. 
+During this period, a user can write continuous frames of data to the transaction. These
+frames must be vertical (i.e. all channels must have a single series specified),
+and all series must have the same number of samples. You can read more about
+frame alignment here.
```
Comment (Lham42) [inline docs/site/src/pages/reference/concepts/writes.mdx:97]: maybe say below instead of here - thought it was referring to a missing link

**Review by Lham42 — CHANGES_REQUESTED** [review]

**Comment by Lham42** [comment]
also go through checklist

File: .vscode/synnax.code-workspace
[the remark this answers is not in the document]
Comment (emilbon99) [reply .vscode/synnax.code-workspace:1]: No. This is important for typescript dev. I did remove the added lines though. They shouldn't be there.

File: client/py/synnax/framer/adapter.py [hunk 3]
```diff
@@ -74,10 +74,29 @@ def update(self, channels: ChannelParams):
...
     def keys(self):
         return self.__keys
 
+    def __adapt_to_key(
+        self,
+        ch: ChannelPayload | ChannelKey | ChannelName
+    ) -> ChannelKey:
+        if isinstance(ch, ChannelKey):
+            return ch
+        if isinstance(ch, ChannelPayload):
```
[the remark this answers is not in the document]
Comment (emilbon99) [reply client/py/synnax/framer/adapter.py:94]: If the caller is using a channel name, we need to resolve the key from the cache or server. Added comment explaining this.

File: client/py/tests/test_frame.py [hunk 4]
```diff
@@ -268,3 +268,9 @@ def test_type_error_when_invalid_value_is_adapted(self, adapter):
         adapter, channel = adapter
         with pytest.raises(TypeError):
             adapter.adapt(Exception("Invalid value"))
+
+    def test_adapt_dict_keys(self, adapter):
+        """Should raise a type error when an invalid value is adapted"""
+        adapter, channel = adapter
+        adapted = adapter.adapt_dict_keys({channel.name: 123})
+        assert adapted[channel.key] == 123
```
[the remark this answers is not in the document]
Comment (emilbon99) [reply client/py/tests/test_frame.py:276]: Forgot to adjust the comment underneath here. It is the valid test. Relevant type errors are covered in other test cases.

File: docs/site/src/pages/reference/concepts/writes.mdx (hunk 2 above)
Comment (emilbon99) [reply docs/site/src/pages/reference/concepts/writes.mdx]: Removed this, and added linear issue https://linear.app/synnaxlabs/issue/SY-1058/document-frames-series-concepts-again to update the frames and series documentation.

**Pull request merged**